pkg/storagev2/migration/migration.go

```diff
@@ -0,0 +1,115 @@
+package migration
+
+import (
+	"encoding/binary"
+	"errors"
+	"fmt"
+	"sort"
+	"unsafe"
+
+	storage "github.com/ethersphere/bee/pkg/storagev2"
+)
+
+type (
+    Version = uint64
+    StepFn func(storage.Store) error
+    StepsMap = map[Version]StepFn
+)
+
+func Migrate(s storage.Store, sm StepsMap) error {
+	if err := ValidateVersions(sm); err != nil {
+		return err
+	}
+
+	currentVersion, err := GetVersion(s)
+	if err != nil {
+		return err
+	}
+
+	for nextVersion := currentVersion + 1; ; nextVersion++ {
+		if stepFn, ok := sm[nextVersion]; ok {
+			err := stepFn(s)
+			if err != nil {
+				return err
+			}
+			err = SetVersion(s, nextVersion)
+			if err != nil {
+				return err
+			}
+		} else {
+			// there is no next version defined
+			// so we stop here
+			return nil
+		}
+	}
+}
+
+// ValidateVersions checks versions if they are in order n (where n min version value), n+1, n+2, n+3... (all values are increasing orders)
+func ValidateVersions(sm StepsMap) error {
+	versions := make([]int, len(sm))
+	i := 0
+	for key := range sm {
+		versions[i] = int(key)
+		i++
+	}
+
+	missing := FindMissingNumbers(versions)
+
+	if len(missing) > 0 {
+		return fmt.Errorf("missing versions: %v", missing)
+	}
+	return nil
+}
```

Comment:
Just realized that this function could be simplified. All we need to do is find `min` and `max` version numbers, and then we can check if all version steps are defined by doing something like this:

```
if (maxVersion - minVersion) == len(sm) {
  // ok
} else {
 // nok
}
```

Reply:
So lets say `[1 2 3]` is the version array

```
3-1 == 3 {
}

```
This is always false.

Reply:
if we tweak formula a bit, like so:
```
(maxVersion - minVersion) == len(sm) - 1 
```

Reply:
Thank worked. Thanks.